### DISCUSSION TITLE
Blazor - Sinalizar status da operação ao acessar uma página com listagem geral de registros

### DISCUSSION THREAD

**Discussion by GCoeCta** [balta-io/forum#468] [Tira Dúvidas]
Caros,

Tenho uma página em uma app Blazor, minha primeira aplicação para estudos a partir do curso do André, na qual apresento ao operador uma listagem geral de registros ( exemplo : cadastro de produtos ).  Esta página é renderizada com @attribute [StreamRendering(true)] de forma a que seja apresentado ao usuário um spinner enquanto a operação é processada.

Na entrada da página eu verifico se há registros a serem mostrados e neste caso apresento uma tabela com os mesmos e não havendo registros mostro o spinner.

Para esta verificação uso esta condição : 

`@if (meusDados.Any()) {
   - mostrar os dados
}
else {
      - mostrar o spinner 
}

Os dados são obtidos através do método protected override async Task OnInitializedAsync().

Funciona perfeitamente desde que haja dados a serem listados. Porém caso não haja informações, não estou conseguindo fazer com que ao invés do spinner seja mostrada uma mensagem como "<h1>Pesquisa sem resultados<h1>".

Neste caso a relação de dados é resultado de uma pesquisa no banco com base em parametros recebidos pela página em uma linha como : 

`meusDados = await meuDb.minhaTabela.Where(x => x.NomeDoImovel.StartsWith(@param1) && x.Uf == @param2).Take(20).ToListAsync();`

Meu procedimento para fazer isto foi após a instrução acima, setar uma variável indicando que o processamento foi realizado e no momento de apresentar o spinner checar se o processamento ocorreu. Tendo sido realizado o processamento eu apresentaria a mensagem de resultados não encontrados. Mas por minha inexperiência em Blazor, não estou tendo sucesso.

Agradeço qq colaboração.
 

`

**Comment by malighieri** [comment]
Eu tenho acompanhado o conteúdo do Balta sobre Blazor, mas ainda não coloquei a mão na massa.
Dentro dos meus conhecimentos de C# e do que entendi do fluxo, eu sugeriria você modificar seu código para o exemplo abaixo. Considerando que a variável meusDados seja inicializada como _**null**_, quando sua busca no banco retornar, ela receberá uma instância que pode ter ou não itens, mas com certeza será diferente de _**null**_.

```
@if (meusDados is not null)
{
     mostrar os dados
}
else
{
     mostrar o spinner
}
```

Completando o fluxo com a possibilidade de não retornar nenhum item, poderia ter o código abaixo:
```
@if (meusDados is not null)
{
    @if (meusDados.Any())
    {
         mostrar os dados
    }
    else
    {
         mostrar mensagem de nenhum dado encontrado
    }
}
else
{
     mostrar o spinner
}
```

**Reply by GCoeCta** [reply]
Olá, tudo bem ?

Tentei sua sugestão. Desta forma, ao invés do spinner a página passa a apresentar a mensagem de "não há dados" em todos os ciclos de processamento, havendo ou não dados.

**Comment by GCoeCta** [comment]
Consegui agora.

Meu fluxo de processamento estava sendo prejudicado pelo fato de haver um timeout nas buscas sem resultado. Ajustei o timeout com esta instrução antes da pesquisa : 

`myDb.Database.SetCommandTimeout(TimeSpan.FromMinutes(20));`
`meusDados = await dbs.Sncrs.Where......ToListAsync();`
`ProcessamentoEncerrado = true;`

E o ciclo ficou assim :  

@if ( (meusDados is not null) && ProcessamentoEncerrado )
{
    @if (meusDados.Any())
    {
         mostrar os dados
    }
    else
    {
         mostrar mensagem de nenhum dado encontrado
    }
}
else
{
     mostrar o spinner
}

Obrigado pela ajuda !
